### PR TITLE
Add build failure RCA patterns

### PR DESCRIPTION
**Pull request by shirangj**
Add build failure RCA patterns in O3DE repo. Build failure RCA lambda functions will read these patterns using Github API.

RFC: https://github.com/o3de/sig-build/blob/main/rfcs/rfc-bld-20220623-1-build-failure-rca.md

### CODE REVIEW

**Comment by brianherrera** [comment]
Just wanted to check the benefits that we get from moving these pattern files from the RCA repo in into the O3DE repo. I recall from the RFC the pattern files were set to be stored in their own repo. 

I remember we discussed also adding validation checks for these files if we stored them in their own repo. This also means changes to these files will need to pass the O3DE AR as well.

File: scripts/build/build_failure_rca/rca_patterns/Build-Environment.json [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+            "log_patterns": [
+                "^Killed part of.*after waiting for.*because we assume unknown agent.*is never going to appear$"
+            ],
+            "log_testcases": [
+                "Killed part of O3DE » PR-9504 #2 after waiting for 5 min 0 sec because we assume unknown agent EC2 (us-west-2) - WIN2019_INC_3 (i-05a91a7e645fc20d1) is never going to appear"
+            ],
+            "owners": []
+        }
+    ]
+}
```
Comment (amzn-changml) [inline scripts/build/build_failure_rca/rca_patterns/Build-Environment.json:88]: Nit: EoL for each json file

**Review by amzn-changml — APPROVED** [review]

**Comment by shirangj** [comment]
> 

We put old RCA patterns in separate repo because Jenkins build failure RCA plugin uses a centralized knowledge base, and it would be hard to update the knowledge base if the RCA patterns are from multiple branches from O3DE.

The benefits of putting new RCA patterns in O3DE repo:
1. We don't need to maintain a separate repo for RCA patterns because build failure RCA lambda functions can read RCA patterns from Github directly.
2. We don't need to redeploy the CDK package when we change RCA patterns.
3. RCA patterns are now tied to O3DE branch, it's more relevant to the build from that branch.

And I agree with the additional check for RCA patterns in PR, I think we can add this check in current validation stage.

**Comment by brianherrera** [comment]
Thanks for providing that info. Agreed that having the patterns attached to the branch will be helpful.

**Review by brianherrera — APPROVED** [review]

**Pull request merged**